recipe/meta.yaml

```
@@ -1,45 +1,47 @@
 {% set name = "cuda-python" %}
-{% set version = "11.8.1" %}
-{% set major_version = version.split(".")[0] %}
-{% set minor_version = version.split(".")[1] %}
-{% set patch_version = version.split(".")[2] %}
+{% set version = "11.8.2" %}
+{% set major_version = version.split(".")[0]|int %}
+{% set minor_version = version.split(".")[1]|int %}
+{% set patch_version = version.split(".")[2]|int %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: ef54ada5d607387f59950cfb3c3852e5816b62824402d289227b8066e8353489
+  sha256: bfb36d314065b695decd23935ab008630cc956cbcd428d520106568fdeb7e644
 
 build:
-  number: 2
+  number: 0
   skip: true  # [py<37 or osx or cuda_compiler_version != "11.8"]
   script:
     - {{ PYTHON }} -m pip install . --no-deps -vv
   run_exports:
     - cuda-python >=11.7.1,<12  # we need this because CUDA Python 11.7.1 has ABI breaks, see conda-forge/cuda-python-feedstock#15
-  ignore_run_exports:
-    - cudatoolkit
+  ignore_run_exports_from:
+    - {{ compiler('cuda') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
     - setuptools
     - cython
     - python
     - pyclibrary
     - pip
+    - cuda-version {{ major_version }}
   run:
     - python
-    - {{ pin_compatible('cudatoolkit', min_pin='x', max_pin='x') }}
+    - {{ pin_compatible('cuda-version', min_pin='x', max_pin='x') }}
+    {% if major is defined and major < 12 %}
+    - cudatoolkit
+    {% endif %}
```

Per offline discussion, there is no way to satisfy `cuda-python` 11.x with a CUDA 12 installation since it looks for particular versions of nvrtc. So we can probably simplify this and rely on the `cuda-version` pinning to constrain to CUDA 11.

Separately, I don't see where `major` is defined (I see `major_version`) so this is probably a bug anyway.
```suggestion
    - cudatoolkit
```

---

`major` is defined based on the `version`, which is `11` up top

In version 12 (with your PR), this branch would effectively be disabled

The idea here was to make it easier to potentially update both versions in `main` (should that be needed)

That said, don't have strong feeling about that

---

I'm not sure if I see that being defined. I see
```
{% set major_version = version.split(".")[0]|int %}
```
but `major_version` is not the same as `major`.

---

I think the ability to update either build in `main` is potentially good, but I don't think I can wrap my head around all the changes that could require once #33 is merged... would prefer to keep it simple for this PR.

---

Sorry you are right. Think it is suppose to be

```suggestion
    {% if major_version < 12 %}
    - cudatoolkit
    {% endif %}
```

Though no strong feelings if we would rather drop

---

My vote is to drop (we can apply my original suggestion above if so).

---

Sounds reasonable. Maybe we can wait for CI to finish. Commit your initial suggestion. Then merge?